### PR TITLE
Support URLs for internal external consumption

### PR DESCRIPTION
**Pull request by sandeep-paliwal**
Added support to return external or internal URLs
Fix for #61 

## Description

getProperties will now return both (default and internal) URLs
generatePresignURL options will support urlType(external | internal) as options to specify the type of URL to return

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x ] I have read the **CONTRIBUTING** document.
- [x ] I have added tests to cover my changes.
- [x ] All new and existing tests passed.

### CODE REVIEW

File: lib/Files.js [hunk 1]
```diff
@@ -39,6 +39,20 @@ function throwNotImplemented (funcName) {
   logAndThrow(new codes.ERROR_NOT_IMPLEMENTED({ messageValues: [funcName] }))
 }
 
+const PUBLIC_URL = 'public'
+const RUNTIME_URL = 'runtime'
```
Comment (purplecabbage) [inline lib/Files.js:43]: nit: just define this in UrlType, there is no need to have 2 ways to get to it.

File: lib/impl/AzureBlobFiles.js [hunk 2]
```diff
@@ -334,10 +335,14 @@ class AzureBlobFiles extends Files {
    * @override
    * @private
    */
-  _getUrl (filePath) {
+  _getUrl (filePath, urlType = UrlType.public) {
```
Comment (purplecabbage) [inline lib/impl/AzureBlobFiles.js:338]: it might make sense to use require('url') here instead of all this string manipulation

**Review by purplecabbage — APPROVED** [review]

**Comment by alexkli** [comment]
@sandeep-paliwal Is there an API doc showing the change? I tried to find it in the tests, but they seem to be working on internal code only (things like `_getPresignedUrl()` look like they are "private").

**Comment by sandeep-paliwal** [comment]
@alexkli I have updated the docs.

**Comment by alexkli** [comment]
hmm, the doc changes seem not really clear.

Is there an example code snippet? I can’t follow how this URLType would be used. What does external and internal mean?

**Comment by meryllblanchet** [comment]
@sandeep-paliwal based on @alexkli , how can we make it clearer what internal / external URLs are used for?
Also could you add some code snippet to the README.md?

**Comment by sandeep-paliwal** [comment]
@alexkli I have updated the Readme [docs](https://github.com/sandeep-paliwal/aio-lib-files/tree/internal_ext_urls#url-types-and-usage), along with usage example.

**Comment by meryllblanchet** [comment]
@sandeep-paliwal , I added some more description to your update.

**Review by alexkli — APPROVED** [review]
Thanks for the doc update, that made it clear. LGTM!

**Pull request merged**